### ISSUE TITLE
Rule proposal: `prefer-logical-assignment`

### ISSUE THREAD

**Issue by zanminkian** [sindresorhus/eslint-plugin-unicorn#2517]
### Description

Morden JavaScript have 3 useful assignment statement:
- `??=`: [Nullish coalescing assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment)
- `||=`: [Logical OR assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR_assignment)
- `&&=`: [Logical AND assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND_assignment)

We can call them `logical assignment`.

Prefer them over `if` and `ternary`.

### Fail

```js
if(foo) foo = "foo";
if(!foo) foo = "foo";
if(foo === undefined) foo = "foo";
if(foo === null) foo = "foo";
if(foo == undefined) foo = "foo";
if(foo == null) foo = "foo";
```

```js
foo = foo ? foo : 'foo';
foo = !foo ? foo : 'foo';
foo = foo !== undefined ? foo : 'foo';
foo = foo !== null ? foo : 'foo';
foo = foo != undefined ? foo : 'foo';
foo = foo != numm ? foo : 'foo';
```

### Pass

```js
foo ||= 'foo';
foo &&= 'foo';
foo ??= 'foo';
```

### Proposed rule name

prefer-logical-assignment

**Comment by fisker** [comment]
[`logical-assignment-operators`](https://eslint.org/docs/latest/rules/logical-assignment-operators)?
